terraform/nr-alerts/nr-alerts.tf

```diff
@@ -0,0 +1,132 @@
+#############################################
+# EC2 Instances
+#############################################
+
+### Fetch Canary EC2 Instances IDs by TAG.
+data "aws_instances" "canary_instances" {
+  filter {
+    name   = "tag:Name"
+    values = [var.instance_name_pattern]
+  }
+}
+
+### Fetch EC2 Instances by IDs.
+data "aws_instance" "canary_instance" {
+  count       = length(data.aws_instances.canary_instances.ids)
+  instance_id = data.aws_instances.canary_instances.ids[count.index]
+}
+
+#############################################
+# Alerts
+#############################################
+
+
+# Prepare alerts name <-> instance name relationship.
+locals {
+  alerts_instances = {
+    for ins in data.aws_instance.canary_instance :
+    ins.tags.Name => can(regex(".+:.+:.+:.+", ins.tags.Name)) ?
+    (join(" / ", [
+      var.policies_prefix,
+      split(":", ins.tags.Name)[3],
+      split(":", ins.tags.Name)[2]
+    ])) :
+    (ins.tags.Name)
+  }
+}
```

Comment:
instead of asking for `instance_name_pattern = "canary:v1.37.2:*"` and then breaking it with a regex to recreate it again...

Why not ask for something like:
```terraform
instance_pattern = {
  name    = "canary"
  version = "v1.37.2"
  distro  = "*"
}
```

Then in the filter above use something like this:
```terraform
data "aws_instances" "canary_instances" {
  filter {
    name   = "tag:Name"
    values = ["${local.name}:${local.version}:${local.distro}"]
  }
}
```

Reply:
This comes from adapting the current on in the agent, and I left it open to not use that patter, but definitely a good advise. I'll check If I can make it more readable 👍